### ISSUE TITLE
Wildcard backtracking

### ISSUE THREAD

**Issue by linkdata** [naoina/denco#2]
The following test code fails:

```
	records = []denco.Record{
		{"/*wildcard", "testroute0"},
		{"/a/:b/c", "testroute1"},
	}
	testcases = []testcase{
		{"/a/b", "testroute0", []denco.Param{{"wildcard", "a/b"}}, true},
		{"/a/b/foo", "testroute0", []denco.Param{{"wildcard", "a/b/foo"}}, true},
		{"/a/b/c", "testroute1", []denco.Param{{"b", "b"}}, true},
		{"/a", "testroute0", []denco.Param{{"wildcard", "a"}}, true},
	}
	runLookupTest(t, records, testcases)
```
`Router.Lookup("/a") => (<nil>, denco.Params(nil), false), want ("testroute0",denco.Params{denco.Param{Name:"wildcard", Value:"a"}}, true)`

I would expect the `/a` lookup to return the `/*wildcard` route. Instead it fails to return anything. Shouldn't the `/a/:b/c` pattern backtrack all the way to the root?

**Comment by naoina** [comment]
That's true. I'll try to fix it because it is unexpected behavior.

**Comment by naoina** [comment]
I have fixed it by commit 7e5f34b675c929608e4edaf4c4b97819e111b929.

**Comment by naoina** [comment]
@linkdata Thank you for report the issue :)
